SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/DeploymentLauncher/DeploymentLauncher.cs

```diff
@@ -541,27 +588,92 @@ private static int CreateSimDeployments(string[] args, bool useChinaPlatform)
-                simDeploymentConfig.RegionCode = regionCode;
+                simDeployment.RegionCode = regionCode;
             }
 
-            simDeploymentConfig.Tag.Add(DEPLOYMENT_LAUNCHED_BY_LAUNCHER_TAG);
-            simDeploymentConfig.Tag.Add(SIM_PLAYER_DEPLOYMENT_TAG);
+            simDeployment.Tag.Add(DEPLOYMENT_LAUNCHED_BY_LAUNCHER_TAG);
+            simDeployment.Tag.Add(SIM_PLAYER_DEPLOYMENT_TAG);
 
             Console.WriteLine(
-                $"Creating the simulated player deployment {simDeploymentName} in project {projectName} with {simNumPlayers} simulated players. Link: https://console.improbable.io/projects/{projectName}/deployments/{simDeploymentName}/overview");
+                $"Creating the simulated player deployment {simDeploymentName} in project {projectName} with {numSimPlayers} simulated players. Link: https://console.improbable.io/projects/{projectName}/deployments/{simDeploymentName}/overview");
 
             var simDeploymentCreateOp = deploymentServiceClient.CreateDeployment(new CreateDeploymentRequest
             {
-                Deployment = simDeploymentConfig
+                Deployment = simDeployment
             });
 
             return simDeploymentCreateOp;
         }
 
+        private static List<Operation<Deployment, CreateDeploymentMetadata>> CreateSimPlayerDeploymentsAsync(DeploymentServiceClient deploymentServiceClient,
+            string projectName, string assemblyName, string runtimeVersion, string mainDeploymentName, string simDeploymentBaseName,
+            string simDeploymentJsonPath, string simDeploymentSnapshotPath, string regionCode, string clusterCode, int numSimPlayers, int numSimDeployments, bool useChinaPlatform)
+        {
+            var operations = new List<Operation<Deployment, CreateDeploymentMetadata>>();
+
+            var longestName = GetSimDeploymentName(simDeploymentBaseName, numSimDeployments);
+            if (longestName.Length > 32)
+            {
+                Console.WriteLine($"The deployment name may not exceed 32 characters. '{longestName}' would have {longestName.Length}.");
+                return operations;
+            }
+
+            for (var simPlayerDeploymentId = 1; simPlayerDeploymentId <= numSimDeployments; ++simPlayerDeploymentId)
+            {
+                var simDeploymentName = simDeploymentBaseName;
+                if (simPlayerDeploymentId != 1)
+                {
+                    simDeploymentName += ("_" + simPlayerDeploymentId);
+                }
+
+                try
+                {
+                    if (DeploymentExists(deploymentServiceClient, projectName, simDeploymentName))
+                    {
+                        StopDeploymentByName(deploymentServiceClient, projectName, simDeploymentName);
+                    }
+
+                    // Determine the amount of simulated players in this deployment
+                    var numDeploymentSimplayers = numSimPlayers / numSimDeployments;
```

Comment:
I wonder if it's better usability to have `numSimPlayersPerDeployment` rather than dividing the total num by the num deploys.

Reply:
I do think that it's a better variable name, so I've changed that now.

Do you mean that we could consider having `numSimPlayersPerDeployment` as a command argument instead of `numSimDeployments`? I chose the latter because, to a customer, the number of deployments (presumably) has a much stronger correlation with cost. In this way they know exactly what they're launching (even if they don't feel like doing the arithmetic :P).

Reply:
My idea would be to have `numDeployments` and `numSimPlayersPerDeployment`. Gives the same data as you suggest but is easier to digest at a glance. No biggie though, my idea also means you have to have numSimPlayers be a multiple of numDeployments so lose some granularity

Reply:
I've had a think and changed things up a little. Users can now provide the optional `maxPlayersPerDeployment` parameter. If the parameter is not provided a single deployment is created for all simplayers. This means that

* There are no nasty surprises, if a user chooses to pass `maxPlayersPerDeployment` they will know that multiple deployments will be started.
* The user can specify the exact number of simplayers, so that they can test their game with the exact number of players they have in mind.
* When scale-testing (and after determining how many simplayers a single deployment supports) they only have to change the `numSimPlayers` argument, as the number of deployments will scale along.

Do let me know what you think about this approach